runtime/src/iree/hal/drivers/hip/hip_device.c

```diff
@@ -1411,12 +1430,26 @@ static iree_status_t iree_hal_hip_device_perform_buffer_operation_now(
               stream, iree_hal_buffer_allocation_size(data->buffer));
           break;
         }
-        status = iree_hal_hip_allocator_alloc_async(
+        if (is_hip_allocator) {
```

Comment:
WE should probably add a comment here saying this "is_hip_allocator" is temporary, and we plan on removing once the caching allocator is timeline aware, because then we can stop differentiating between types of allocator.